compiler/arm/codegen/ARMBinaryEncoding.cpp

```diff
@@ -69,17 +69,14 @@ uint32_t encodeHelperBranchAndLink(TR::SymbolReference *symRef, uint8_t *cursor,
 
 uint32_t encodeHelperBranch(bool isBranchAndLink, TR::SymbolReference *symRef, uint8_t *cursor, TR_ARMConditionCode cc, TR::Node *node, TR::CodeGenerator *cg)
    {
-   int32_t  distance;
-   uint32_t target;
+   intptrj_t target = (intptrj_t)symRef->getMethodAddress();
 
-   target = (uintptr_t)symRef->getMethodAddress();
-   distance = target - (uintptr_t) cursor;
-   if (distance > BRANCH_FORWARD_LIMIT || distance < BRANCH_BACKWARD_LIMIT)
+   if (cg->directCallRequiresTrampoline(target, (intptrj_t)cursor))
       {
       target = cg->fe()->indexedTrampolineLookup(symRef->getReferenceNumber(), (void *)cursor);
-      TR_ASSERT(((int32_t)target - (int32_t)cursor) <= BRANCH_FORWARD_LIMIT &&
-             ((int32_t)target - (int32_t)cursor) >=  BRANCH_BACKWARD_LIMIT,
-             "CodeCache is more than 32MB.\n");
+
+      TR_ASSERT(TR::Compiler->target.cpu.isTargetWithinBranchImmediateRange(target, (intptrj_t)cursor),
+                "Target address is out of range.  CodeCache is more than 32MB.");
```

Comment:
It seems the asserting condition does not have any indication that CodeCache must not be more than 32MB. It may be more clear removing "CodeCache is more than 32MB."

Reply:
Sure.